### ISSUE TITLE
Items to add based on model examples

### ISSUE THREAD

**Issue by dsyme** [DiffSharp/DiffSharp#227]
Looking through some models and I notice this is used in one of them.  I don't know its relative importance, but noting it for completeness and tracking.

TorchSharp:

* [x] avgpool1d, 2d, 3d and reverse mode for these (TorchSharp)

* [x] activation functions gelu, silu, hardswish, relu6, hardsigmoid (TorchSharp)

* [x] permute (TorchSharp )

* [x] split based on count (TorchSharp)

* [ ] UpSampling1d, 2d, 3d (TorchSharp)

DiffSharp:

* [x] avgpool1d, 2d, 3d and reverse mode for these, done pending merge, see #252 

* [x] permute (DiffSharp) See #193, done pending merge, see #254 

* [ ] activation functions gelu, silu, hardswish, relu6, hardsigmoid

* [x] split based on count 

* [ ] LayerNorm functions and model

* [ ] mean/sum/stddev of multiple dimensions (was #216)

* [ ] DepthwiseConv2d

Other things to consider:

* [ ] RMSProp optimizer https://pytorch.org/docs/stable/optim.html#torch.optim.RMSprop

* [ ] AdaDelta optimizer 

* [ ] GlobalAvgPool2d model

* [ ] UpSampling2d model 

* [x] MaxPool1d/2d/3d model

* [ ] ZeroPadding2d function and model

* [ ] randn giving mean and stddev

* [ ] Embedding

* [x] Use TorchSharp loss functions (binary_cross_entropy etc.) 

* [x] max/min along dimensions (was #232)
